temporian/implementation/numpy/data/event.py

```diff
@@ -163,9 +172,18 @@ def from_dataframe(
         )
 
         # convert timestamp column to float
-        df[timestamp_column] = df[timestamp_column].apply(
-            convert_date_to_duration
-        )
+        # TODO: This is taking a lot of time. Don't use apply.
+
+        # if timestamp column is of kind int convert to float
```

Comment:
Pandas can represent duration / timestamps with a type (e.g. int64) and a unit (e.g. s, µs, ns).
Are we sure this code works for all units?

Reply:
Sure.

```python
import numpy as np
a = np.array([1490195805], dtype='datetime64[s]')
b = np.array([1490195805433502912]).astype('datetime64[ns]')

print(f"a:{a}")
print(f"b:{b}")

print(f"a.astype(int):{a.astype(int)}")
print(f"b.astype(int):{b.astype(int)}")
```

Results:

```
a:['2017-03-22T15:16:45']
b:['2017-03-22T15:16:45.433502912']
a.astype(int):[1490195805]
b.astype(int):[1490195805433502912]
```

Both "a" and "b", represent the same date. However, when casted to integer, the value is different because "a" represent the date in seconds since the epoche, while "b" is the time in ns since the epoche. When converting to temporian, we want all the dates to be a float64 in seconds.

I think the same exist in pandas types:

```python
a = pd.to_datetime(1490195805, unit='s')
b = pd.to_datetime(1490195805433502912, unit='ns')
```

Reply:
Okay so the user in the API will be able to pass any object as datetime and we will parse them with the function `pd,.to_datetime()` independent of what the user passes as an object (unless it's a plain int or float).

<img width="1233" alt="image" src="https://user-images.githubusercontent.com/33181424/234097507-fc8790f5-c613-4c18-9717-d13cd82ee11f.png">

This is the output of what will happen if the user uses two datetimes with different units. In the end, both will have the same timestamp float but one will have more precision.